### PR TITLE
LT-1682: signature for C&C files - RSA

### PR DESCRIPTION
**Pull request by andrey-anokhin**

### CODE REVIEW

File: src/MarginTrading.CommissionService.SqlRepositories/MarginTrading.CommissionService.SqlRepositories.csproj [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <Version>1.10.7</Version>
   </PropertyGroup>
   <ItemGroup>
+    <PackageReference Include="BouncyCastle.NetCore" Version="1.8.5" />
```
Comment (GSailen) [inline src/MarginTrading.CommissionService.SqlRepositories/MarginTrading.CommissionService.SqlRepositories.csproj]: What about license?

File: src/MarginTrading.CommissionService.SqlRepositories/Repositories/CostsAndChargesRepository.cs [hunk 2]
```diff
@@ -146,12 +151,17 @@ private CostsAndChargesEntity Map(CostsAndChargesCalculation calculation)
...
-                Data = data
+                Data = signedData
             };
         }
 
         private CostsAndChargesCalculation Map(CostsAndChargesEntity entity)
         {
+            if (!RsaHelper.ValidateSign(entity.Data, _settings.SignatureSettings, out var error))
+            {
+                throw new Exception($"Document with id {entity} has invalid signature. Error: {error}");
```
Comment (GSailen) [inline src/MarginTrading.CommissionService.SqlRepositories/Repositories/CostsAndChargesRepository.cs]: entity.Id?

File: src/MarginTrading.CommissionService.SqlRepositories/RsaHelper.cs [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+using MarginTrading.CommissionService.Core.Settings;
+using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Crypto.Generators;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.OpenSsl;
+using Org.BouncyCastle.Security;
+
+namespace MarginTrading.CommissionService.SqlRepositories
+{
+    public static class RsaHelper
```
Comment (GSailen) [inline src/MarginTrading.CommissionService.SqlRepositories/RsaHelper.cs]: Is all this some kinda copy-paste?

File: src/MarginTrading.CommissionService.SqlRepositories/MarginTrading.CommissionService.SqlRepositories.csproj (hunk 1 above)
Comment (andrey-anokhin) [reply src/MarginTrading.CommissionService.SqlRepositories/MarginTrading.CommissionService.SqlRepositories.csproj]: README (https://github.com/chrishaly/bc-csharp/blob/master/README.md) points to http://www.bouncycastle.org/licence.html, so MIT as I understand.

File: src/MarginTrading.CommissionService.SqlRepositories/Repositories/CostsAndChargesRepository.cs (hunk 2 above)
Comment (andrey-anokhin) [reply src/MarginTrading.CommissionService.SqlRepositories/Repositories/CostsAndChargesRepository.cs]: Yes )

File: src/MarginTrading.CommissionService.SqlRepositories/RsaHelper.cs (hunk 3 above)
Comment (andrey-anokhin) [reply src/MarginTrading.CommissionService.SqlRepositories/RsaHelper.cs]: Some kinda, but not 100%

**Pull request merged**